libcxx/include/__math/hypot.h

```diff
@@ -9,11 +9,16 @@
 #ifndef _LIBCPP___MATH_HYPOT_H
 #define _LIBCPP___MATH_HYPOT_H
 
+#include <__algorithm/max.h>
 #include <__config>
+#include <__math/abs.h>
+#include <__math/roots.h>
 #include <__type_traits/enable_if.h>
 #include <__type_traits/is_arithmetic.h>
 #include <__type_traits/is_same.h>
 #include <__type_traits/promote.h>
+#include <array>
```

Comment:
Sorry for not spotting this earlier, but I would like to avoid including `<array>` here. This is a low-level math function and we should avoid pulling in `<array>`, which then pulls in a ton of additional stuff (like some algorithms we totally don't need here).

It's kinda hacky, but we could just define our own `__hypot_factors_pair` type as a struct with two members instead.

@philnik777 WDYT about introducing something like this to break dependencies for internal uses only (not as part of this patch though)?

```c++
template <class _Tp, class _Up>
struct __basic_pair {
  [[no_unique_address]] _Tp __first;
  [[no_unique_address]] _Up __second;
};
```

Reply:
`__utility/pair.h` is quite lean by now I think. It's mostly depending on type traits and (since C++20) concepts. `std::pair` is also used in quite a few public interfaces, so I'm not sure we'd gain much from introducing another `pair`.

Reply:
Fixed in cf378ed2d4dd28d8bb5e50ba66c61e30e2195994. I replaced `std::array` usage by `std::pair`.